inference-engine/thirdparty/clDNN/kernel_selector/core/cl_kernels/resample_opt.cl

```diff
@@ -53,11 +77,161 @@ KERNEL (resample_opt)(__global INPUT0_TYPE* input,
+#else
+    const int batch = (int)get_global_id(2) % OUTPUT_BATCH_NUM;
+#endif
+
+    ACCUMULATOR_TYPE i_b = AXES_USED[0] ? FUNC_CALL(get_original_coordinate)(batch, SCALES[0], out_size[0], PADDED_B) : batch;
+    ACCUMULATOR_TYPE i_f = AXES_USED[1] ? FUNC_CALL(get_original_coordinate)(feature, SCALES[1], out_size[1], PADDED_F) : feature;
+    ACCUMULATOR_TYPE i_y = AXES_USED[3] ? FUNC_CALL(get_original_coordinate)(oy, SCALES[3], out_size[3], PADDED_Y) : oy;
+    ACCUMULATOR_TYPE i_x = AXES_USED[4] ? FUNC_CALL(get_original_coordinate)(ox, SCALES[4], out_size[4], PADDED_X) : ox;
+
+#if PADDING_USED == 1
+    i_b -= PADS_BEGIN[0];
+    i_f -= PADS_BEGIN[1];
+    i_y -= PADS_BEGIN[3];
+    i_x -= PADS_BEGIN[4];
+#endif
+
+    const int ib_r = (int)i_b;
+    const int if_r = (int)i_f;
+    const int iy_r = (int)i_y;
+    const int ix_r = (int)i_x;
+
+#if ANTIALIAS == 1
+    const ACCUMULATOR_TYPE ab = 1.0f / SCALES[0];
+    const ACCUMULATOR_TYPE af = 1.0f / SCALES[1];
+    const ACCUMULATOR_TYPE ay = 1.0f / SCALES[3];
+    const ACCUMULATOR_TYPE ax = 1.0f / SCALES[4];
+
+    const int rb = (SCALES[0] < 1.0f) ? 2 : (int)ceil(TO_ACCUMULATOR_TYPE(KERNEL_W) / ab);
+    const int rf = (SCALES[1] < 1.0f) ? 2 : (int)ceil(TO_ACCUMULATOR_TYPE(KERNEL_W) / af);
+    const int ry = (SCALES[3] < 1.0f) ? 2 : (int)ceil(TO_ACCUMULATOR_TYPE(KERNEL_W) / ay);
+    const int rx = (SCALES[4] < 1.0f) ? 2 : (int)ceil(TO_ACCUMULATOR_TYPE(KERNEL_W) / ax);
+#else
+    const ACCUMULATOR_TYPE ab = 1.0f;
+    const ACCUMULATOR_TYPE af = 1.0f;
+    const ACCUMULATOR_TYPE ay = 1.0f;
+    const ACCUMULATOR_TYPE ax = 1.0f;
+
+    const int rb = (SCALES[0] < 1.0f) ? 1 : (int)ceil(TO_ACCUMULATOR_TYPE(KERNEL_W) / ab);
+    const int rf = (SCALES[1] < 1.0f) ? 1 : (int)ceil(TO_ACCUMULATOR_TYPE(KERNEL_W) / af);
+    const int ry = (SCALES[3] < 1.0f) ? 1 : (int)ceil(TO_ACCUMULATOR_TYPE(KERNEL_W) / ay);
+    const int rx = (SCALES[4] < 1.0f) ? 1 : (int)ceil(TO_ACCUMULATOR_TYPE(KERNEL_W) / ax);
+#endif
+
+    int const b_init = max(-PADS_BEGIN[0], ib_r - rb);
+    int const f_init = max(-PADS_BEGIN[1], if_r - rf);
+    int const y_init = max(-PADS_BEGIN[3], iy_r - ry);
+    int const x_init = max(-PADS_BEGIN[4], ix_r - rx);
+
+    int const b_max = min(PADS_END[0] + INPUT0_BATCH_NUM, ib_r + rb + 1);
+    int const f_max = min(PADS_END[1] + INPUT0_FEATURE_NUM, if_r + rf + 1);
+    int const y_max = min(PADS_END[3] + INPUT0_SIZE_Y, iy_r + ry + 1);
+    int const x_max = min(PADS_END[4] + INPUT0_SIZE_X, ix_r + rx + 1);
+
+#ifndef LEFTOVERS
+    const int fp_max = FEATURE_BLOCK_SIZE;
+#else
+    const int fp_max = min(FEATURE_BLOCK_SIZE, FEATURE_LEFTOVER);
+#endif
+
+    ACCUMULATOR_TYPE wb = ACCUMULATOR_VAL_ZERO;
+    ACCUMULATOR_TYPE wf = ACCUMULATOR_VAL_ZERO;
+    ACCUMULATOR_TYPE wy = ACCUMULATOR_VAL_ZERO;
+    ACCUMULATOR_TYPE wx = ACCUMULATOR_VAL_ZERO;
+    ACCUMULATOR_TYPE w  = ACCUMULATOR_VAL_ZERO;
+
+    unroll_for(int fp = 0; fp < fp_max; fp+=16) {
```

Comment:
If it is not intended to use `unroll`, could you just use `for` instead?

Reply:
replaced to for